hippod/garbage_handler.py

```diff
@@ -0,0 +1,195 @@
+                    if achiev['id'] == achiev_id:
+                        del data['achievements'][i]
+            with open(path, 'w') as data_file:
+                data = json.dumps(data, sort_keys=True,indent=4,
+                      separators=(',', ': '))
+                data_file.write(data)
+
+
+        def correct_container(app, sha_major, sha_minor):
+            path = os.path.join(app['DB_OBJECT_PATH'],
+                        sha_major[0:2],
+                        sha_major,
+                        'container.db')
+            with open(path) as data_file:
+                data = json.load(data_file)
+                for i, sc in enumerate(data['subcontainer-list']):
+                    if sc['sha-minor'] == sha_minor:
+                        del data['subcontainer-list'][i]
+            with open(path, 'w') as data_file:
+                data = json.dumps(data, sort_keys=True,indent=4,
+                      separators=(',', ': '))
+                data_file.write(data)
+
+
+        def correct_object_index(app, sha_major):
+            path = os.path.join(app['DB_OBJECT_PATH'],
+                        'object-index.db')
+            with open(path) as data_file:
+                data = json.load(data_file)
+                for i, d in enumerate(data):
+                    if d['object-item-id'] == sha_major:
+                        del data[i]
+            with open(path, 'w') as data_file:
+                data = json.dumps(data, sort_keys=True,indent=4,
+                      separators=(',', ': '))
+                data_file.write(data)            
+
+
+        def check_achiev_lifetime(self, app, sha_major, sha_minor, sc_content, sc_list):
+            achiev_list = list()
+            for achiev in sc_content['achievements']:
+                achiev_id = str(achiev['id'])
+                date_added = achiev['date-added']
+                achievement = hippod.api_shared.get_achievement_data_by_sha_id(app, sha_major, sha_minor, achiev_id)
+                date_added = datetime.datetime.strptime(date_added, '%Y-%m-%dT%H:%M:%S.%f')
+                diff = datetime.datetime.now() - date_added
+                diff = GarbageHandler.GarbageHandlerCollect.get_sec(diff)
+                if 'anchor' in achievement and diff > self.lifetime_anchored:
+                    path = os.path.join(sha_major[0:2], sha_major, sha_minor, 'achievements', '{}.db'.format(achiev_id))
+                    achiev_list.append(path)
+                    GarbageHandler.GarbageHandlerCollect.correct_subcontainer(app, sha_major, sha_minor, achiev)
+                    if len(sc_content['achievements']) == 1:
+                        path = os.path.join(sha_major[0:2], sha_major, sha_minor)
+                        achiev_list.append(path)
+                        GarbageHandler.GarbageHandlerCollect.correct_container(app, sha_major, sha_minor)
+                        if len(sc_list) == 1:
```

Comment:
same, what happens if != 1, raise an error()? Looks like failures are not properly catched somehow